src/cpp/fastrtps_deprecated/publisher/PublisherHistory.cpp

```diff
@@ -355,3 +401,23 @@ bool PublisherHistory::get_next_deadline(
 
     return false;
 }
+
+bool PublisherHistory::is_key_registered(
+        const InstanceHandle_t& handle)
+{
+    if (mp_writer == nullptr || mp_mutex == nullptr)
+    {
+        logError(RTPS_HISTORY, "You need to create a Writer with this History before using it");
+        return false;
+    }
+    std::lock_guard<RecursiveTimedMutex> guard(*this->mp_mutex);
+    t_m_Inst_Caches::iterator vit;
+    vit = keyed_changes_.find(handle);
+    return (vit != keyed_changes_.end() &&
+            (vit->second.cache_changes.empty() ||
+                (NOT_ALIVE_UNREGISTERED != vit->second.cache_changes.back()->kind &&
+                NOT_ALIVE_DISPOSED_UNREGISTERED != vit->second.cache_changes.back()->kind
+                )
+            )
+           );
```

Comment:
Empty means registered?

Reply:
Yes, if writer is configured as VOLATILE, the vector can be empty.

Reply:
And it could also have been just registered. Ok.